raiden-ts/src/services/epics.ts

```diff
@@ -164,200 +173,37 @@ export const pathFindServiceEpic = (
-                if (error_code === 2201) {
-                  throw new RaidenError(ErrorCodes.PFS_NO_ROUTES_BETWEEN_NODES);
-                }
-
-                throw new RaidenError(ErrorCodes.PFS_ERROR_RESPONSE, {
-                  errorCode: error_code,
-                  errors,
-                });
-              }
-              const filteredPaths: Paths = [],
-                invalidatedRecipients = new Set<Address>();
-              // eslint-disable-next-line prefer-const
-              for (let { path, fee } of data.paths) {
-                // if route has us as first hop, cleanup/shift
-                if (path[0] === deps.address) path = path.slice(1);
-                const recipient = path[0];
-                // if this recipient was already invalidated in a previous iteration, skip
-                if (invalidatedRecipients.has(recipient)) continue;
-                // if we already found some valid route, allow only new routes through this peer
-                const canTransferOrReason = !filteredPaths.length
-                  ? channelCanRoute(
-                      state,
-                      presences,
-                      action.meta.tokenNetwork,
-                      recipient,
-                      action.meta.target,
-                      action.meta.value.add(fee) as UInt<32>,
-                    )
-                  : recipient !== filteredPaths[0].path[0]
-                  ? 'path: already selected another recipient'
-                  : fee.gt(filteredPaths[0].fee)
-                  ? 'path: already selected a smaller fee'
-                  : true;
-                if (canTransferOrReason !== true) {
-                  log.warn(
-                    'Invalidated received route. Reason:',
-                    canTransferOrReason,
-                    'Route:',
-                    path,
-                  );
-                  invalidatedRecipients.add(recipient);
-                  continue;
-                }
-                filteredPaths.push({ path, fee });
-              }
-              if (!filteredPaths.length) throw new RaidenError(ErrorCodes.PFS_NO_ROUTES_FOUND);
-              yield pathFind.success({ paths: filteredPaths }, action.meta);
-            })(),
-          ),
-        ),
-        catchError((err) => of(pathFind.failure(err, action.meta))),
+        mergeMap((latest) => {
+          const { target } = action.meta;
+          let presenceRequest: Observable<matrixPresence.request>;
+          let latestWithTargetInPresences: Observable<Latest>;
+
+          if (target in latest.presences) {
+            presenceRequest = EMPTY;
+            latestWithTargetInPresences = of(latest);
+          } else {
+            presenceRequest = of(matrixPresence.request(undefined, { address: target }));
+            latestWithTargetInPresences = waitForMatrixPresenceResponse$(action$, deps, target);
+          }
+
+          return merge(
+            presenceRequest,
```

Comment:
I'd suggest you put the request in the end of `merge`. Just to be safe, in case the synchronous calls would respond to your request before your observable below would be subscribed.

Reply:
:heavy_check_mark: